### PR TITLE
feat: allow setting development mode when creating an OIDC app

### LINKED ISSUE #6397: Allow setting development mode when creating an application
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

When I want to setup a client with local configuration, then I have to enable the development mode. Enabling the development mode is only possible after creating the application, requiring additional steps from a user after setup.

Currently only a warning is shown that the user needs to enable development mode. Development mode can't be directly enabled.

### Describe your ideal solution

When I add an insecure redirect url / post-logout url, 
Then I get a warning that this requires development mode enabled AND
Then I can toggle development mode on the same page

Should be possible both in Wizard Mode and Pro Mode.
Screenshot 2023-08-19 at 10 46 05
Screenshot 2023-08-19 at 10 46 13

### Version

_No response_

### Environment

ZITADEL Cloud

### Additional Context

Current "Workaround": Create the application, then select the application, navigate to Redirect Settings, then toggle Development Mode

### PR DESCRIPTION
**Pull request by doncicuto**
As requested in #6397, this PR adds the following functionality for OIDC apps:

- The development mode toggle is added to the app creation wizard and pro mode

Wizard:

image

Pro mode:

image

- The development mode boolean is passed to the redirect-uris component so the toggle changes classes applied to redirect uris

If toggle is not checked, warning is shown

image

If toggle is checked, no warning is shown

image

- In the wizard, in the final stage I've added a Development Mode entry to show if it's enabled or disabled

image

Closes #6397 

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by hifabienne** [comment]
Hei @doncicuto 
I am sorry I absolutely missed this pr. we will have a look at it

**Comment by doncicuto** [comment]
Hi @peintnermax, I'll try to fix the conflicts today or tomorrow

**Comment by doncicuto** [comment]
Conflicts fixed @peintnermax

File: console/src/app/pages/projects/apps/app-create/app-create.component.html [hunk 1]
```diff
@@ -439,6 +463,14 @@ <h1>{{ 'APP.PAGES.CREATE_DESC_TITLE' | translate }}</h1>
         "
       >
         <div class="formfield full-width">
+          <mat-slide-toggle
+            color="primary"
+            name="devMode"
+            matTooltip="{{ 'APP.OIDC.DEVMODEDESC' | translate }}"
+            (change)="onChangeDevMode($event)"
```
Comment (peintnermax) [inline console/src/app/pages/projects/apps/app-create/app-create.component.html:470]:
```suggestion
            (change)="onChangeDevMode($event)"
```

Comment (peintnermax) [inline console/src/app/pages/projects/apps/app-create/app-create.component.html:470]: lets use [(ngModel)]="devMode" here so we need no change handler
probably we also have to add `[ngModelOptions]="{ standalone: true }"`

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts [hunk 2]
```diff
@@ -328,6 +331,11 @@ export class AppCreateComponent implements OnInit, OnDestroy {
     }
   }
 
+  public onChangeDevMode(ob: MatLegacySlideToggleChange) {
+    this.devMode = ob.checked;
+    this.oidcAppRequest.setDevMode(this.devMode);
+  }
+
```
Comment (peintnermax) [inline console/src/app/pages/projects/apps/app-create/app-create.component.ts:338]:
```suggestion
```

remove this if we use `ngModel`

**Review by peintnermax — CHANGES_REQUESTED** [review]
Hi @doncicuto thx for your nice contribution.. 👍  I've made some suggestions on how to have a little cleaner code..

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts (hunk 2 above)
Comment (doncicuto) [reply console/src/app/pages/projects/apps/app-create/app-create.component.ts:338]: Thanks @peintnermax for you review as always and sorry if I make silly mistakes, my trip from React to Angular sometimes has issues :smile: 

I see no problem about using ngModel to keep things clean, but where do you suggest I should add 

`this.oidcAppRequest.setDevMode(this.devMode)`

so the devMode value is added to the request and sent when the new app is created. Should I add a set/get instead?

Comment (peintnermax) [reply console/src/app/pages/projects/apps/app-create/app-create.component.ts:338]: no problem. thx for your effort as always.. 👍 
I did totally oversee that sorry. We also use redirectUris as ngModel but define the setter and getter classes which directly set the value to our `oidcAppRequest`

```
public get devMode() {
    return this.oidcAppRequest.toObject().devMode;
  }

  public set devMode(value: boolean) {
    this.oidcAppRequest.setDevMode(value);
  }
```

then we can also remove the global variable

Comment (doncicuto) [reply console/src/app/pages/projects/apps/app-create/app-create.component.ts:338]: Perfect @peintnermax , that will be the approach then, give me some minutes and I'll fix the code and do some testing

**Review by peintnermax — APPROVED** [review]
nice job thx a lot 👍

**Pull request merged**